app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -45,6 +47,23 @@ def self.default_capsule
         end
       end
 
+      def puppet_path
+        self[:puppet_path] || update_puppet_path
+      end
+
+      def update_puppet_path
+        return unless has_feature?(PULP_NODE_FEATURE)
```

Comment:
Ahhh, okay I see now. I hadn't understood what was using the pulp feature. I'll go ahead and trying deduplicating the two.